packages/discovery-react-components/src/components/DocumentPreview/DocumentPreview.tsx

```diff
@@ -60,17 +65,61 @@ const DocumentPreview: FC<Props> = ({
     // setLoading(true);
   }, [doc]);
 
+  // fetch PDF (if necessary)
+  useEffect(() => {
+    async function fetchFile() {
+      const hasFile = await documentProvider!.provides(document!);
+      if (hasFile) {
+        setProvidedFile(await documentProvider?.get(document!));
+      }
+    }
+
+    // `file` takes precedence over a file provided by `documentProvider`
+    if (file) {
+      setProvidedFile(file);
+    } else if (document && documentProvider) {
+      fetchFile();
+    } else {
+      setProvidedFile(undefined);
+    }
+  }, [document, documentProvider, file]);
+
+  // TODO Seems like these 2 useEffects should just be part of PdfViewer
+  const [textMappings, setTextMappings] = useState<TextMappings | null>(null);
+  useEffect(() => {
+    const mappings = getTextMappings(doc);
+    if (mappings) {
+      setTextMappings(mappings);
+    }
+  }, [doc]);
+
+  // Pull total page count from either the PDF file or the structural
+  // data list
+  const [pageCount, setPageCount] = useState(0);
+  const [pdfPageCount, setPdfPageCount] = useState(pageCount);
+  useEffect(() => {
+    if (providedFile && pdfPageCount > 0) {
+      setPageCount(pdfPageCount);
+    } else if (textMappings) {
+      const last = textMappings.text_mappings.length - 1;
+      setPageCount(textMappings?.text_mappings[last].page.page_number ?? 1);
+    }
+  }, [textMappings, providedFile, pdfPageCount]);
```

Comment:
This code is only used by the `PdfViewer` -- it should be refactored to a better location in the code (maybe within `PdfViewer` itself.

Reply:
For future, we could be able to allows _page count_ even in HTML/TEXT preview. But for now, I agree it's only for PDF. So I move the code to `PdfViewer`.